File: src/problems_all/217_contains_duplicate/contains_duplicate.py
"""
Runtime:            584 ms
Beats:              5.04%
Memory:             26.1 MB
Beats:              95.39%
Submission:         https://leetcode.com/problems/contains-duplicate/submissions/678186913/
Time complexity:    O(n)
Space complexity:   O(n)
Topics:             #array, #hash-table, #sorting
Solved By:          #hash-table
"""

from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_set = set(nums)
        return len(nums) != len(num_set)


def main():
    nums = [1, 2, 3, 1]
    result = Solution().containsDuplicate(nums)
    print(result)


if __name__ == '__main__':
    main()

File: src/problems_all/217_contains_duplicate/contains_duplicate.go
/*
Runtime:            105 ms
Beats:              5.39%
Memory:             9.6 MB
Beats:              96.17%
Submission:         https://leetcode.com/problems/contains-duplicate/submissions/881242084/
Time complexity:    O(n)
Space complexity:   O(n)
Topics:             #array, #hash-table, #sorting
Solved By:          #hash-table
*/

package main

import "fmt"

func main() {
	nums := []int{1, 2, 3, 4}
	result := containsDuplicate(nums)
	fmt.Println(result)
}

func containsDuplicate(nums []int) bool {
	counter := map[int]int8{}
	for _, num := range nums {
		if _, ok := counter[num]; ok {
			return true
		} else {
			counter[num] = 1
		}
	}

	return false
}
